### ISSUE TITLE
API for visuals's color changing

### ISSUE THREAD

**Issue by exor2008** [vispy/vispy#1764]
Right now not all visuals seems to have standard and obvious way to change it's color in runtime.

For example if you want to change color to green you could do

`mycolor = (0.0, 1.0, 0.0)`

and then for `scene.Line`:
`line.set_data(color=mycolor) # the limit of dreams`

for `scene.Markers`:
`marker.set_data(pos=mypos, face_color=mycolor, edge_color=mycolor) #have to specify pos everytime`

for `scene.AxisWidget`:
```
axis.axis.axis_color = mycolor
axis.axis.tick_color = mycolor
axis.axis._text.color = mycolor
axis.axis._axis_label.color = mycolor
axis.axis._need_update = True
```

and most unobvious one I've came across, for `scene.GridLines`:
```
from vispy.color import Color
grid._grid_color_fn['color'] = Color(mycolor).rgba
```

So, it's gonna be cool to have one clear way to change visuals' color, for example an universal `set_data` method. It also could be a bunch of properties like `color, bgcolor, border_color, text_color` withch already presented in some widgets.

**Comment by djhoese** [comment]
The last one with GridLines, as @kmuehlbauer pointed out, is a private attribute of the class and was his suggestion since there is no interface for setting it otherwise.

The hard part for implementing this is deciding what exactly "color" means for each Visual. Do we provide a `.color` that does everthing in the "foreground" but then allow other color properties for sub-setting (ticks have one color, labels have another, etc).

**Comment by exor2008** [comment]
You can have a set of specific properties if needed, like `fgcolor, bgcolor, axis_color` for complex visuals and just `color` for simple cases. And duplicate all properties' names in `set_data`

**Comment by kmuehlbauer** [comment]
@djhoese Would that be a topic for the dev meeting? At least we could discuss possible next steps.

**Comment by djhoese** [comment]
@kmuehlbauer Yes, we could. This would be something @rougier @almarklein and @rossant should look into with their new CZI grant.

**Comment by djhoese** [comment]
Based on discussion at today's meeting, we support AxisWidget being updated with public properties for each sub-component. We should update Visuals to have color properties with names `color` (for single color objects), `.edge_color` and `.face_color` for objects with two colors. If we can identify Visuals that don't fit this simple structure of one or two colors or sub-components with their own colors.
